travis-ci/install-gnutls.sh should exit early if libnettle can't be installed

I noticed an error in [a recent Travis CI log](https://travis-ci.org/clojure-emacs/cider/jobs/324610574)

```
configure: error: 
  ***
  *** Libnettle 3.1 was not found.
```

## Expected behavior

`install-gnutls.sh` should download and compile (1) libnettle, and (2) gnutls.

## Actual behavior

When libnettle isn't found (in the example job linked above, it looks like this particular Travis CI worker got borked somehow) this install script carries on and attempts to compile gnutls anyway, but it obviously can't succeed without the necessary dependency.

## Steps to reproduce the problem

This is frustrating, but the error is transient and seems to be relatively rare. The faulty behavior is clear from the logs, however. Simple fix.

## Environment & Version information

n/a

### Lein/Boot version

n/a

### Emacs version

n/a

### Operating system

n/a

--- a/travis-ci/install-gnutls.sh
+++ b/travis-ci/install-gnutls.sh
@@ -38,6 +38,8 @@ curl -O https://ftp.gnu.org/gnu/nettle/nettle-${NETTLE_VERSION}.tar.gz \
     && make -j4 install \
     && make distclean
 
+if [ "$?" -ne "0" ]; then exit 1; fi
+
 cd $WORKDIR
 curl -O https://www.gnupg.org/ftp/gcrypt/gnutls/v3.5/gnutls-${GNUTLS_VERSION}.tar.xz \
     && xz -d -k gnutls-${GNUTLS_VERSION}.tar.xz \

